diff -Naur faiss-1.7.1/_build_python.orig/loader.py faiss-1.7.1/_build_python/loader.py 
--- faiss-1.7.1/_build_python.orig/loader.py      2021-07-17 17:39:39.000000000 +0100
+++ faiss-1.7.1/_build_python/loader.py 2021-07-19 09:24:02.000000000 +0100
@@ -62,5 +62,13 @@
 if not has_AVX2:
     # we import * so that the symbol X can be accessed as faiss.X
     logger.info("Loading faiss.")
-    from .swigfaiss import *
+#    from .swigfaiss import *
+    import ctypes
+    dll = ctypes.PyDLL('_swigfaiss.so')
+    init = dll.PyInit__swigfaiss
+    init.restype = ctypes.py_object
+    module = init()
+    for attr in dir(module):
+        if attr[0] != '_':
+            globals()[attr] = getattr(module, attr)
     logger.info("Successfully loaded faiss.")
     